src/DataProvider/SerializerAwareDataProviderTrait.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\DataProvider;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Inject serializer in data providers.
+ *
+ * @author Vincent Chalamon <vincent@les-tilleuls.coop>
+ */
+trait SerializerAwareDataProviderTrait
```

Comment:
This is related to https://github.com/api-platform/core/pull/1184#issuecomment-311166174 as a serializer alias or alternative.